log/internal_logger.go

```diff
@@ -30,30 +31,45 @@ func SetAnalyticsServerURL(url string) {
 }
 
 // Internal sends the log message to the configured analytics server
-func (lm Entry) Internal(stepID, tag string, data map[string]interface{}) {
-	lm.Data = make(map[string]interface{})
+func (e Entry) Internal(stepID, tag string, data map[string]interface{}) {
+	e.Data = make(map[string]interface{})
 	for k, v := range data {
-		lm.Data[k] = v
+		e.Data[k] = v
 	}
 
-	if v, ok := lm.Data["step_id"]; ok {
+	if v, ok := e.Data["step_id"]; ok {
 		fmt.Printf("internal logger: data.step_id (%s) will be overriden with (%s) ", v, stepID)
 	}
-	if v, ok := lm.Data["tag"]; ok {
+	if v, ok := e.Data["tag"]; ok {
 		fmt.Printf("internal logger: data.tag (%s) will be overriden with (%s) ", v, tag)
 	}
 
-	lm.Data["step_id"] = stepID
-	lm.Data["tag"] = tag
+	e.Data["step_id"] = stepID
+	e.Data["tag"] = tag
 
 	var b bytes.Buffer
-	if err := json.NewEncoder(&b).Encode(lm); err != nil {
+	if err := json.NewEncoder(&b).Encode(e); err != nil {
 		return
 	}
 
-	go func() {
-		if _, err := netClient.Post(analyticsServerURL + "/logs", "application/json", &b); err != nil {}
-	}()
+	ctx, cancel := context.WithCancel(context.TODO())
+	_ = time.AfterFunc(3 * time.Second, func() {
```

Comment:
You need to close the timer as soon as you know that it won't need to be fired. Hopefully this won't cause any error, but what if the request is already done and a left over timer fires and calls `cancel` that might already will result in a null pointer exception? Not sure how it works, but in case you no need to fire the timer anymore, please stop and destroy it.

Reply:
what about this:

```
	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
	defer cancel() // Canceling this context releases resources associated with it, so code should call cancel as soon as the operations running in this Context complete:

	req, err := http.NewRequest(http.MethodPost, analyticsServerURL+"/logs", &b)
	if err != nil {
		// deliberately not writing into users log
		return
	}

	req = req.WithContext(ctx)

	req.Header.Add("Content-Type", "application/json")

	if _, err := httpClient.Do(req); err != nil {
		// deliberately not writing into users log
		return
	}
```